### ISSUE TITLE
`atoms_to_metadata` schema fails when the `Atoms` object has a dummy atom

### ISSUE THREAD

**Issue by kumaranu** [Quantum-Accelerators/quacc#480]
I am running transition state calculation for the following molecular system using Sella (version 2.3.2).

```
20

C   2.5130658   1.8776065   1.33016696    
C   2.16854858  1.22687201  0.00040975    
C   1.66613644 -0.10224323  0.02112403    
O  -0.12419935  0.3006476   0.81627268    
C  -0.78490488  1.05660144  0.11259288    
O  -0.25398334  2.06119629 -0.59100452    
C  -2.28830635  0.88566987 -0.0503305     
C  -2.62680382 -0.52470317 -0.54245261    
H   1.72994847  1.7267763   2.04601596    
H   2.70023361  2.93198613  1.22023455    
H   3.4360011   1.41862281  1.73888855    
H   1.07239067  1.90948762 -0.38813635    
H   2.91281418  1.45745234 -0.7656041     
H   1.32374389 -0.51238627 -0.88236216    
H   2.14096316 -0.79807226  0.67390623    
H  -2.76816731  1.05898887  0.91637478    
H  -2.66689684  1.62777492 -0.73408853    
H  -2.22643938 -1.27045915  0.13982672    
H  -3.70358155 -0.65795186 -0.60756258    
H  -2.20140947 -0.70329162 -1.52574925    
```
My optimization looks like:
```
2023-06-22 00:51:46,000 INFO Hostname/IP lookup (this will take a few seconds)
2023-06-22 00:51:46,006 INFO Launching Rocket
2023-06-22 07:51:52,556 INFO RUNNING fw_id: 5502 in directory: /global/scratch/users/kumaranu/ts_jobs/prod6/block_2023-06-15-03-17-02-056843/launcher_2023-06-22-07-49-23-121347
2023-06-22 07:51:52,576 INFO Task started: {{jobflow.managers.fireworks.JobFiretask}}.
2023-06-22 07:51:52,584 INFO Starting job - ts_job (24e97bc3-743d-4948-9b56-e60be02b9e66)
     Step     Time          Energy         fmax         cmax       rtrust          rho
Sella   0 00:52:04      -83.111551       3.7807       0.0000       0.1000       1.0000
Sella   1 00:53:13      -83.132821       1.6308       0.0000       0.1000       1.7928
Sella   2 00:53:24      -83.247212       0.9493       0.0000       0.1000       1.6924
Sella   3 00:53:36      -83.354530       0.8286       0.0000       0.1000       1.1131
Sella   4 00:53:48      -83.454893       0.6912       0.0000       0.1150       1.0048
Sella   5 00:54:00      -83.546907       0.5362       0.0000       0.1322       0.9797
Bad internals found!
Bad internals found!
Sella   6 00:54:13      -83.577724       0.3185       0.0000       0.1322       1.0000
Sella   7 00:55:52      -83.699512       0.4404       0.0000       0.1322       1.0939
Sella   8 00:56:05      -83.760889       0.3846       0.0000       0.1322       0.9365
Sella   9 00:56:17      -83.805950       0.2618       0.0000       0.1322       1.0622
Sella  10 00:56:29      -83.832877       0.2694       0.0000       0.1521       1.0323
Sella  11 00:56:41      -83.845451       0.2129       0.0000       0.1521       0.7344
Sella  12 00:56:53      -83.846740       0.1626       0.0000       0.1521       0.3751
Sella  13 00:57:05      -83.848936       0.0674       0.0000       0.1521       1.0627
Sella  14 00:57:17      -83.850640       0.0731       0.0000       0.1521       1.3268
Sella  15 00:57:29      -83.853673       0.0961       0.0000       0.1521       1.4722
Sella  16 00:57:41      -83.856334       0.0985       0.0000       0.1521       1.1861
Sella  17 00:57:52      -83.858311       0.1026       0.0000       0.1521       1.5838
Sella  18 00:58:05      -83.863291       0.0757       0.0000       0.1521       1.2602
Sella  19 00:58:17      -83.868475       0.1464       0.0000       0.1521       1.0572
Sella  20 00:58:29      -83.872239       0.1854       0.0000       0.1521       1.1081
Sella  21 00:58:42      -83.875704       0.1587       0.0000       0.1521       1.0423
Sella  22 00:58:55      -83.876954       0.0797       0.0000       0.1521       0.6726
Sella  23 00:59:07      -83.878304       0.0807       0.0000       0.1521       1.2960
Sella  24 00:59:18      -83.879430       0.0794       0.0000       0.1521       1.5094
Sella  25 00:59:30      -83.880084       0.0472       0.0000       0.1521       1.2817
Sella  26 00:59:41      -83.880323       0.0294       0.0000       0.1521       1.0441
Sella  27 00:59:52      -83.880406       0.0194       0.0000       0.1521       1.4711
Sella  28 01:00:04      -83.880459       0.0074       0.0000       0.1521       1.3025
2023-06-22 08:00:04,667 INFO Rocket finished
```
My error looks like:
```
No GPU/TPU found, falling back to CPU. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)
/global/home/users/kumaranu/Documents/NewtonNet/newtonnet/data/loader.py:72: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  result["R"] = torch.tensor(input['R'],
/global/home/users/kumaranu/Documents/NewtonNet/newtonnet/data/loader.py:76: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  result["Z"] = torch.tensor(input['Z'],
Traceback (most recent call last):
  File "/global/home/users/kumaranu/.conda/envs/quacc/lib/python3.9/site-packages/fireworks/core/rocket.py", line 261, in run
    m_action = t.run_task(my_spec)
  File "/global/home/users/kumaranu/.conda/envs/quacc/lib/python3.9/site-packages/jobflow/managers/fireworks.py", line 162, in run_task
    response = job.run(store=store)
  File "/global/home/users/kumaranu/.conda/envs/quacc/lib/python3.9/site-packages/jobflow/core/job.py", line 544, in run
    response = function(*self.function_args, **self.function_kwargs)
  File "/global/home/users/kumaranu/.conda/envs/quacc/lib/python3.9/site-packages/covalent/_workflow/electron.py", line 754, in wrapper
    return electron_object(*args, **kwargs)
  File "/global/home/users/kumaranu/.conda/envs/quacc/lib/python3.9/site-packages/covalent/_workflow/electron.py", line 375, in __call__
    return self.function(*args, **kwargs)
  File "/global/home/users/kumaranu/.conda/envs/quacc/lib/python3.9/site-packages/monty/dev.py", line 113, in decorated
    return _callable(*args, **kwargs)
  File "/global/home/users/kumaranu/quacc/quacc/recipes/newtonnet/core.py", line 199, in ts_job
    ts_summary = summarize_opt_run(dyn, additional_fields={"name": "NewtonNet TS"})
  File "/global/home/users/kumaranu/quacc/quacc/schemas/ase.py", line 314, in summarize_opt_run
    "trajectory": [
  File "/global/home/users/kumaranu/quacc/quacc/schemas/ase.py", line 315, in <listcomp>
    atoms_to_metadata(atoms, charge_and_multiplicity=charge_and_multiplicity)
  File "/global/home/users/kumaranu/quacc/quacc/schemas/atoms.py", line 124, in atoms_to_metadata
    metadata = MoleculeMetadata().from_molecule(mol).dict()
  File "/global/home/users/kumaranu/.conda/envs/quacc/lib/python3.9/site-packages/emmet/core/structure.py", line 284, in from_molecule
    symmetry = PointGroupData.from_molecule(meta_molecule)
  File "/global/home/users/kumaranu/.conda/envs/quacc/lib/python3.9/site-packages/emmet/core/symmetry.py", line 70, in from_molecule
    pga = PointGroupAnalyzer(
  File "/global/home/users/kumaranu/.conda/envs/quacc/lib/python3.9/site-packages/pymatgen/symmetry/analyzer.py", line 945, in __init__
    self.centered_mol = mol.get_centered_molecule()
  File "/global/home/users/kumaranu/.conda/envs/quacc/lib/python3.9/site-packages/pymatgen/core/structure.py", line 3367, in get_centered_molecule
    center = self.center_of_mass
  File "/global/home/users/kumaranu/.conda/envs/quacc/lib/python3.9/site-packages/pymatgen/core/structure.py", line 3002, in center_of_mass
    wt = site.species.weight
  File "/global/home/users/kumaranu/.conda/envs/quacc/lib/python3.9/site-packages/pymatgen/core/composition.py", line 479, in weight
    return Mass(sum(amount * el.atomic_mass for el, amount in self.items()), "amu")
  File "/global/home/users/kumaranu/.conda/envs/quacc/lib/python3.9/site-packages/pymatgen/core/composition.py", line 479, in <genexpr>
    return Mass(sum(amount * el.atomic_mass for el, amount in self.items()), "amu")
  File "/global/home/users/kumaranu/.conda/envs/quacc/lib/python3.9/site-packages/pymatgen/core/periodic_table.py", line 1436, in __getattr__
    raise AttributeError(a)
AttributeError: atomic_mass
```

I believe it is due to the dummy atoms with atomic symbol 'X' that are added during the optimization process to deal with the internal coordinates reset in Sella.
The ase Atoms object in the Sella transition state trajectory is passed back with an extra atom for example (a simple example only for illustration):```

Iteration 0: "H2O"
Iteration 1: "H2O"
Iteration 2: "H2OX"
Iteration 3: "H2OX", etc
```

**Comment by arosen93** [comment]
I will first try to explain the error and then suggest a way you can go ahead and solve it.

When Quacc tries to make a summary of the run, it converts the ASE `Atoms` object to a Pymatgen `Molecule`, which is then converted into the schema (dictionary)you're used to seeing. However, the traceback throws an error because Pymatgen is unable to find the atomic mass for an element "X" since, naturally, X is not a real element and does not have a mass.

The example below reproduces this behavior independent of Quacc:
```python
from ase.build import molecule
from pymatgen.io.ase import AseAtomsAdaptor

atoms = molecule("CH4")
atoms[-1].symbol = "X"
molecule = AseAtomsAdaptor().get_molecule(atoms,charge_spin_check=False)
molecule[-1].species.weight
```

This is not really a bug with Pymatgen, although that's where the cause of the issue is. There is simply no way for Pymatgen to report the mass of an element that doesn't exist.

So, what to do? My personal suggestion would be to modify `quacc.schemas.atoms.atoms_to_metadata` to strip any dummy atoms off before the metadata is constructed.

https://github.com/Quantum-Accelerators/quacc/blob/6f445cd97777bd0d245751ae201edfe98f3e26d2/quacc/schemas/atoms.py#L116-L124

In other words, add a line between `structure` and `metadata` as well as between `mol` and `metadata` that strips off any dummy atoms from the Pymatgen `Structure` or `Molecule` object so that the metadata can be made appropriately.

**Comment by kumaranu** [comment]
This problem has been resolved. Thanks @arosen93.
